docs-guides/source/stateful-models.md

```diff
@@ -227,6 +227,219 @@ potential runtime performance improvements.
 For instance, please check out the [2024 WWDC session](https://developer.apple.com/videos/play/wwdc2024/10159/) for an 
 example that uses the [Mistral 7B model](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.2)
 and utilizes the stateful prediction feature for improved performance on a GPU on a macbook pro.  
+The code for converting and deploying Mistral 7B model is released in [Hugging Face Mistral7B Example](
+https://github.com/huggingface/swift-transformers/blob/preview/Examples/Mistral7B/export.py) along with the
+[blog article](https://huggingface.co/blog/mistral-coreml).
 
+## Example: Toy Attention Model with Stateful KV-Cache
 
+To help you better understand how to make an attention model stateful with key cache and value cache, here is a
+toy example.
 
+We start with a toy model with simple attention, where the `query`, `key`, and `value` are calculated by a linear layer
+and then fed into the `scaled_dot_product_attention`.
+This toy example only focus on stateful kv-cache, so it omits some other details such as multi-head, multi-layer,
+positional encoding, final logits, etc.
+```python
+import torch
+import torch.nn as nn
+
+class SimpleAttention(nn.Module):
+     def __init__(self, embed_size):
+         super().__init__()
+         self.query = nn.Linear(embed_size, embed_size)
+         self.key = nn.Linear(embed_size, embed_size)
+         self.value = nn.Linear(embed_size, embed_size)
+
+     def forward(self, x):
+         Q = self.query(x)  # (batch_size, seq_len, embed_size)
+         K = self.key(x)  # (batch_size, seq_len, embed_size)
+         V = self.value(x)  # (batch_size, seq_len, embed_size)
+         return torch.nn.functional.scaled_dot_product_attention(Q, K, V)
+
+class ToyModel(nn.Module):
+     def __init__(self, vocab_size, embed_size):
+         super().__init__()
+         self.embedding = nn.Embedding(vocab_size, embed_size)
+         self.attention = SimpleAttention(embed_size)
+         self.fc = nn.Linear(embed_size, embed_size)
+
+     def forward(self, x):
+         embedded = self.embedding(x)
+         attention_output = self.attention(embedded)
+         return self.fc(attention_output)
+```
+
+To use key cache and value cache in the attention, we can write a new class `SimpleAttentionWithKeyValueCache` which
```

Comment:
"in the attention" is kind of strange. Maybe "for attention" or "in the attention model".

Reply:
Changed.